sagemathinc/cocalc
# start supporting expiring PII info

# Description
* a new server setting `pii_retention`
* if not `never`,  a little helper to derive number of seconds from it + display it
* then uses this number to create an expiration date in the future (using that specific misc function)
* and sets an expiration date when inserting into the `central_log` , iff the type of event is of a certain kind or contains `email_address` or `ip_address`
* **bonus** if the server settings are set, the cache is reset

# Testing Steps
* What I did is to just restart the hub (nothing set) and check if it works → no expiration
* then set to 30 days → expires in the future
* I checked if these strings give a meaningful results via a quick test

# Relevant Issues

### [Checklist](https://github.com/sagemathinc/cocalc/wiki/PR-Checklist):

Front end:
